sklearn/tests/test_multioutput.py

```diff
@@ -102,6 +147,63 @@ def test_multi_target_sample_weights():
 n_samples, n_features = X.shape
 n_outputs = y.shape[1]
 n_classes = len(np.unique(y1))
+classes = list(map(np.unique, (y1, y2, y3)))
+
+
+def test_multi_output_classification_partial_fit_parallelism():
+    sgd_linear_clf = SGDClassifier(loss='log', random_state=1)
+    mor = MultiOutputClassifier(sgd_linear_clf, n_jobs=-1)
+    mor.partial_fit(X, y, classes)
+    est1 = mor.estimators_[0]
+    mor.partial_fit(X, y)
+    est2 = mor.estimators_[0]
+    # parallelism requires this to be the case for a sane implementation
+    assert_false(est1 is est2)
+
+
+def test_multi_output_classification_partial_fit():
+    # test if multi_target initializes correctly with base estimator and fit
+    # assert predictions work as expected for predict
+
+    sgd_linear_clf = SGDClassifier(loss='log', random_state=1)
+    multi_target_linear = MultiOutputClassifier(sgd_linear_clf)
+
+    # train the multi_target_linear and also get the predictions.
+    half_index = X.shape[0] // 2
+    multi_target_linear.partial_fit(
+        X[:half_index], y[:half_index], classes=classes)
+
+    predictions = multi_target_linear.predict(X)
+    assert_equal((n_samples, n_outputs), predictions.shape)
+
+    # train the linear classification with each column and assert that
+    # predictions are equal after first partial_fit
+    for i in range(3):
+        # create a clone with the same state
+        sgd_linear_clf = clone(sgd_linear_clf)
+        sgd_linear_clf.partial_fit(
+            X[:half_index], y[:half_index, i], classes=classes[i])
+        assert_array_equal(sgd_linear_clf.predict(X), predictions[:, i])
+
+    multi_target_linear.partial_fit(X[half_index:], y[half_index:])
+    predictions = multi_target_linear.predict(X)
+    # train the linear classification with each column and assert that
+    # predictions are equal after two partial_fit
+    for i in range(3):
+        # create a clone with the same state
+        sgd_linear_clf = clone(sgd_linear_clf)
+        sgd_linear_clf.partial_fit(
+            X[:half_index], y[:half_index, i], classes=classes[i])
+        sgd_linear_clf.partial_fit(X[half_index:], y[half_index:, i])
+        assert_array_equal(sgd_linear_clf.predict(X), predictions[:, i])
```

Comment:
Could you try and put both these blocks in a loop? It'd save us some redundancy in code.

(sorry for the nagging)